### PR TITLE
Deprecation warnings: print module name if line number is 0

### PR DESCRIPTION
**Pull request by timholy**
Hmm. That was surprisingly easy.

### CODE REVIEW

**Comment by timholy** [comment]
Hmm, but it doesn't always print the *right* module. Any tips?

**Comment by StefanKarpinski** [comment]
cc @JeffBezanson

**Comment by JeffBezanson** [comment]
What I would try is passing a module argument to `jl_binding_deprecation_warning`. From `jl_get_global`, this would be the module argument to that function. From codegen, it would be the module from the `jl_codectx_t`.

**Comment by timholy** [comment]
Thanks for the suggestion! The package I could reliably trigger this on previously doesn't compile on 0.7 right now, so I'm not sure what to do to test this. Merge and see if anyone complains about inaccuracy? Or leave the branch here until someone happens to find a good test case?

File: src/module.c [hunk 1]
```diff
@@ -489,10 +489,12 @@ JL_DLLEXPORT int jl_is_binding_deprecated(jl_module_t *m, jl_sym_t *var)
     return b && b->deprecated;
 }
 
+jl_value_t *jl_get_current_module(void);
```
Comment (vtjnash) [inline src/module.c]: seems to be unused? (this function is deprecated)

**Review by vtjnash — APPROVED** [review]

**Comment by timholy** [comment]
I can confirm that this alters #25397 so that you get
```julia
julia> WARNING: Base.Void is deprecated, use Nothing instead.
 in module DataStructures
```
At least that way you know which package to start checking.

**Comment by quinnj** [comment]
Unfortunately I still see 
```julia
┌ Warning: `using A: B` will only be allowed for single bindings, not modules. Use `using A.B` instead
│   caller = ip:0x0
└ @ Core :-1
```
on this branch wrt #25397.

**Comment by vtjnash** [comment]
> module from the jl_codectx_t

Inlining may make this misleading, there's accurate information in `ctx.builder.getCurrentDebugLocation()`, but I don't know if you can stringify it fully directly, or if you need to convert that to a `DILocation`  and print out the line number information from it (we have a printer for them, but it's embedded in disasm.cpp::LineNumberAnnotatedWriter::emitInstructionAnnot).

**Comment by vtjnash** [comment]
Many deprecations go through `jl_depwarn`, which just calls `depwarn` in julia. (depwarn in Julia only fully supports being called from `@noinline` functions in Julia – it doesn't support being used from C)

**Comment by timholy** [comment]
I was hasty, I was looking at a different depwarn than the one in #25397. On master the same depwarn is
```julia
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near no file:0
```
so at least this is progress :smile:.

**Comment by timholy** [comment]
Interestingly, this only helps with loading the precompiled package. (The warnings issued during building of the `*.ji` file have useful line numbers and filenames.) But if you don't fix them when you first use the package, afterwards it's hard to figure out which package(s) need fixing unless you `touch` all of your packages.

**Comment by c42f** [comment]
Huh, I've got some code which looks exactly like this (ie, passing the module into `jl_binding_deprecation_warning`) over on the `cjf/runtime-logging` branch.  The main goal there is to put the binding deprecations through the logging system, but adding the module ended up being part of it.  I couldn't figure out how to add more precise information (ie, file:line) without major changes to track the origin of a binding.

**Comment by timholy** [comment]
Interesting. Should I merge this, or are you going to have it covered as part of something more ambitious? (Is there an open PR?)

**Comment by c42f** [comment]
Well, I've put up a WIP over at #25413.  Merging this will just cause conflicts there, but it wouldn't be a disaster.  Also, I'm not sure the approach there will fly quite as it is.

**Comment by timholy** [comment]
Apologies for forcing you to fix merge conflicts, but given that #25413 looks like it may have some design issues, I'll go ahead and merge this.

**Comment by c42f** [comment]
No problem, thanks TIm.

**Pull request merged**